src/game_state/MovingObject.h

```diff
@@ -0,0 +1,35 @@
+#ifndef MOVING_OBJECT_
+#define MOVING_OBJECT_
+
+#include <vector>
+#include <glm/glm.hpp>
+#include <memory>
+
+#include "GameObject.h"
+#include "Shape.h"
+
+class MovingObject {
+ public:
+
+  virtual float GetVelocity() {
+    return velocity;
+  }
+
+  virtual std::shared_ptr<std::vector<glm::vec3>> GetPath() {
+    return path;
+  }
```

Comment:
This wasn't supposed to be on this branch but the vec3s are points that the object travels between forming a circuit